morango/sync/session.py

```diff
@@ -47,12 +64,8 @@ def prepare_request(self, request):
         """
         prepped = super(SessionWrapper, self).prepare_request(request)
 
-        try:
-            content_length = prepped.headers.get("Content-Length", 0)
-            if content_length:
-                self.bytes_sent += int(content_length)
-        except TypeError:
-            pass
+        self.bytes_sent += _length_of_headers(prepped.headers)
+        self.bytes_sent += _headers_content_length(prepped.headers)
```

Comment:
No we aren't assuming it's always set. If the request has no body, the requests helper will not add it. So I believe this should be okay.

Reply:
Right, sorry, what I mean is: can we safely assume that *if* the request has a body, the Content-Length header will have already been added, so that we wouldn't miss counting that transfer amount? I know on the receiving side we handle the case of the missing header (in case it was... removed by a proxy or something?), but here I'm guessing it's OK, as we have more control (request is being initiated locally).

Reply:
The missing header on the response side is for when the response is chunked. The request body can be chunked too, but for the sync, none of the requests are chunked because we give the whole body. So if there is a body, then it would have the content length header. I can add comments so that's clear.